generator/src/lib.rs

```diff
@@ -1578,31 +1581,26 @@ pub fn derive_setters(_struct: &vkxml::Struct) -> Option<Tokens> {
         })
     });
 
-    let mut nexts = Vec::new();
     let extends_name = name_to_tokens(&format!("Extends{}", name));
-    if let Some(extends) = &_struct.extends {
-        for target in extends.split(',') {
-            let target = match target {
-                // https://github.com/KhronosGroup/Vulkan-Docs/pull/870
-                "VkPhysicalDeviceProperties" => "VkPhysicalDeviceProperties2",
-                x => x,
-            };
-            let target_ident = name_to_tokens(&format!("Extends{}", name_to_tokens(target)));
-            nexts.push(quote! {
-                unsafe impl #target_ident for #name {}
-            });
-        }
-    }
 
-    let next_function = if has_next {
+    let next_extends = _struct.extends.as_ref().map(|extends| {
+        let extends = extends
+            .split(',')
+            .find(|extend| root_create_info_names.contains(&extend.to_string()))
+            .expect("Should have a root create info");
+        name_to_tokens(&format!("Extends{}", name_to_tokens(&extends)))
```

Comment:
Shouldn't we be implementing the extension trait of *every* struct that this struct extends, rather than just an arbitrary one?

Reply:
What do you mean by arbitrary one? Only root create infos will create a new trait, and only **non** root create infos that extend a root create info will implement this trait.

Reply:
Unless I'm missing something, this seems to be used for deciding when to *implement* the trait, not when to define it. A single extension struct (e.g. `VkPhysicalDeviceVariablePointerFeatures`) can extend multiple root structs, and must implement all of their extension traits.

Reply:
Ah yes that is what I wanted to implement. I just chose the first one.